lib/rules/require-spaces-in-for-statement.js

```diff
@@ -58,22 +58,19 @@ module.exports.prototype = {
 
     check: function(file, errors) {
         file.iterateNodesByType('ForStatement', function(node) {
-            var prevToken;
             if (node.test) {
-                var testToken = file.getTokenByRangeStart(node.test.range[0]);
-                prevToken = file.getPrevToken(testToken);
+                var testToken = file.getFirstNodeToken(node.test);
                 errors.assert.whitespaceBetween({
-                    token: prevToken,
+                    token: file.getPrevToken(testToken),
```

Comment:
How are we solving this for all other rules - maybe that should be part of the whitespaceBetween method? I don't know how common it is to have comments in for loop statements